src/algorithms/all_paths.h

```diff
@@ -12,6 +12,9 @@
 #include "../graph/graph.h"
 #include "../../deps/GraphBLAS/Include/GraphBLAS.h"
 
+#undef MIN
```

Comment:
Why do you undefine MIN?

Reply:
That's a good question - the simple answer is that this line was necessary in order to use a macro named `MIN`.

Without the `UNDEF`, I get the compiler warning:
`/home/laptop-linux/dev/redis/graph/src/execution_plan/ops/../../algorithms/all_paths.h:15:0: warning: "MIN" redefined`

And without the macro entirely, I get;
```
 /home/laptop-linux/dev/redis/graph/src/algorithms/all_paths.c:95:28: warning: implicit declaration of function ‘MIN’ [-Wimplicit-function-declaration]
unsigned int pathLen = MIN(16, maxLen - minLen);
```

This seems to be a rather standard problem - this exact approach is actually used in `deps/GraphBLAS/Demo/Include/demos.h`.

I got the impression from doing some research that this is question is a bit of a can of worms. To my mind, the 2 easiest safe options are keeping this line as it is, or renaming the macro so that we don't shadow the potential MIN definition.